src/cluster/kv/util/lock_test.go

```diff
@@ -634,6 +635,81 @@ func TestWatchAndUpdateTime(t *testing.T) {
 	leaktest.Check(t)
 }
 
+func TestWatchAndUpdateDuration(t *testing.T) {
```

Comment:
Lotttt of copy pasta here--any chance of doing something about that as part of this? Also I haven't read it closely, but `time.Sleep` in threading code always makes me think it's going to flake ;).